pkg/tcpip/stack/ndp.go

```diff
@@ -720,6 +743,83 @@ func (ndp *ndpState) handleRA(ip tcpip.Address, ra header.NDPRouterAdvert) {
 	}
 }
 
+func (ndp *ndpState) sendRouterAdvert() {
+	ref := ndp.nic.getRefOrCreateTemp(header.IPv6ProtocolNumber, header.IPv6AllRoutersMulticastAddress, NeverPrimaryEndpoint, forceSpoofing)
+	la := tcpip.Address(net.ParseIP("fe80::2")) //
```

Comment:
RAs MUST be sent from a link-local address of the NIC the RA is being sent from. As per RFC 4861 section 4.2 https://tools.ietf.org/html/rfc4861#section-4.2.

If we do not have an assigned link-local address, then we can't send an RA

Sending it from a static `fe80::2` will make nodes on the link assume there is a router with link-local address `fe80::2` which may not be true, potentially resulting in a network that is unable to talk to the global internet.

Reply:
Ok, I changed it to ndp.nic.stack.GetMainNICAddress(ndp.nic.ID(), header.IPv6ProtocolNumber), and check it is header.IsV6LinkLocalAddress(). Is that good?